app/components/high-charts.js

```diff
@@ -1,204 +0,0 @@
-import { assign } from '@ember/polyfills';
-import Component from '@ember/component';
-import { getOwner } from '@ember/application';
-import { set, computed } from '@ember/object';
-import { run } from '@ember/runloop';
-import merge from 'deepmerge';
-
-let localConfig = null;
-let defaultOptions = {
-  plotOptions: {
-    series: {
-      shadow: false,
-    },
-  },
-
-  global: {
-    timezoneOffset: new Date().getTimezoneOffset(),
-  },
-
-  credits: {
-    enabled: false,
-  },
-};
-
-export function getSeriesMap(seriesGroup) {
-  let seriesMap = seriesGroup.reduce((seriesMap, seriesItem) => {
-    seriesMap[seriesItem.name] = seriesItem;
-    return seriesMap;
-  }, {});
-
-  return seriesMap;
-}
-
-export function getSeriesChanges(contentSeries, series) {
-  let updatedKeys = Object.keys(contentSeries).filter((key) => {
-    let isValidKey = key !== 'data' && key.charAt(0) !== '_';
-    let isValidType =
-      ['object', 'function'].indexOf(typeof contentSeries[key]) === -1;
-    let isTheSame = contentSeries[key] === series[key];
-
-    return isValidKey && isValidType && !isTheSame;
-  });
-
-  // returns a list of updated keys
-  return updatedKeys;
-}
-
-function setDefaultHighChartOptions(owner, Highcharts) {
-  if (!localConfig) {
-    // use options defined in highcharts-configs/application.js if they exist
-    let configFactory = owner.factoryFor('highcharts-config:application');
-
-    if (configFactory && configFactory.class) {
-      let localConfigBuilder = configFactory.class;
-      localConfig = localConfigBuilder(defaultOptions);
-    } else {
-      localConfig = defaultOptions;
-    }
-  }
-
-  Highcharts.setOptions(localConfig);
-}
-
-/* Map ember-highcharts modes to Highcharts methods
- * https://api.highcharts.com/class-reference/Highcharts.html
- */
-const CHART_TYPES = {
-  StockChart: 'stockChart',
-  Map: 'mapChart',
-  undefined: 'chart',
-};
-
-export default Component.extend({
-  classNames: ['highcharts-wrapper'],
-  content: undefined,
-  mode: undefined,
-  chartOptions: undefined,
-  chart: null,
-  theme: undefined,
-  callback: undefined,
-
-  buildOptions: computed('chartOptions', 'content.[]', function () {
-    let theme = this.theme ?? {};
-    let passedChartOptions = this.chartOptions ?? {};
-
-    let chartOptions = merge(theme, passedChartOptions);
-    let chartContent = this.content;
-
-    // if 'no-data-to-display' module has been imported, keep empty series and leave it to highcharts to show no data label.
-    if (!this.content?.length && !this.highcharts.Chart.prototype.showNoData) {
-      chartContent = [
-        {
-          id: 'noData',
-          data: 0,
-          color: '#aaaaaa',
-        },
-      ];
-    }
-
-    let defaults = { series: chartContent };
-
-    return assign(defaults, chartOptions);
-  }),
-
-  didReceiveAttrs() {
```

Comment:
The code in `didReceiveAttr` hook was unnecessary. I think the code was meant to take care of (1) state management for chart rendering and (2) re-rendering the chart should the input data change (the data never changes in our app).